src/eth_provider/error.rs

```diff
@@ -98,16 +105,29 @@ impl std::fmt::Display for EthApiError {
 /// Constructs a JSON-RPC error object, consisting of `code` and `message`.
 impl From<EthApiError> for ErrorObject<'static> {
     fn from(value: EthApiError) -> Self {
+        (&value).into()
+    }
+}
+
+/// Constructs a JSON-RPC error object, consisting of `code` and `message`.
+impl From<&EthApiError> for ErrorObject<'static> {
+    fn from(value: &EthApiError) -> Self {
         let msg = format!("{value}");
-        let code = EthRpcErrorCode::from(&value);
+        let code = EthRpcErrorCode::from(value);
         let data = match value {
-            EthApiError::Execution(ExecutionError::Evm(EvmError::Other(ref b))) => Some(b),
+            EthApiError::Execution(ExecutionError::Evm(EvmError::Other(ref b))) => Some(b.clone()),
             _ => None,
         };
         ErrorObject::owned(code as i32, msg, data)
     }
 }
 
+impl ToRpcError for EthApiError {
+    fn to_rpc_error(&self) -> ErrorObject<'static> {
```

Comment:
Yes it is required by:

```rust
impl From<EthApiError> for RethEthApiError {
    fn from(value: EthApiError) -> Self {
        Self::other(value)
    }
}
```

Otherwise:

```
error[E0277]: the trait bound `eth_provider::error::EthApiError: ToRpcError` is not satisfied
   --> src/eth_provider/error.rs:50:21
    |
50  |         Self::other(value)
    |         ----------- ^^^^^ the trait `ToRpcError` is not implemented for `eth_provider::error::EthApiError`
    |         |
    |         required by a bound introduced by this call
    |
note: required by a bound in `reth_rpc_eth_types::EthApiError::other`
   --> /Users/tcoratger/.cargo/git/checkouts/reth-36d3ea1d1152b20c/d599393/crates/rpc/rpc-eth-types/src/error.rs:137:21
    |
137 |     pub fn other<E: ToRpcError>(err: E) -> Self {
    |                     ^^^^^^^^^^ required by this bound in `EthApiError::other`
```